service/frontend/workflow_handler.go

```diff
@@ -3723,6 +3833,10 @@ func (wh *WorkflowHandler) PollWorkflowExecutionUpdate(
 		request.WaitPolicy = &updatepb.WaitPolicy{}
 	}
 
+	if err := validateExecution(request.UpdateRef.WorkflowExecution); err != nil {
+		return nil, err
+	}
```

Comment:
NOTE: Added this to ensure that the `SetAttributes` call will not crash, but this is useful/needed anyway.